### DISCUSSION TITLE
Can checkpoint_callback still be used to pass in a predefined ModelCheckpoint?

### DISCUSSION THREAD

**Discussion by andrei-rusu** [Lightning-AI/pytorch-lightning#8354] [General]
Hi all,

Not sure if this is a bug or I'm just missing something, but it seems like I am no longer able to pass a ModelCheckpoint to the checkpoint_callback parameter, in the latest 1.3.8 version. Passing True works as expected, but a predefined checkpoint seems to be ignored and defaulted to None. Example code:

```Python
from pytorch_lightning.callbacks import ModelCheckpoint
checkpoint_callback = ModelCheckpoint(
    monitor='val_loss',
    mode='min',
)
trainer = pl.Trainer(checkpoint_callback=checkpoint_callback)
print(trainer.checkpoint_callback) # This is None
```

**Comment by ananthsub** [comment]
No, the model checkpoint callback should be passed in as part of the `callbacks` argument in the Trainer. `checkpoint_callback` is a boolean only indicating whether or not to save checkpoints
